src/compiler/checker.ts

```diff
@@ -4220,8 +4227,11 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         }
     }
 
-    function reportNonExportedMember(node: Node, name: Identifier, declarationName: string, moduleSymbol: Symbol, moduleName: string): void {
-        const localSymbol = tryCast(moduleSymbol.valueDeclaration, canHaveLocals)?.locals?.get(name.escapedText);
+    function reportNonExportedMember(node: Node, name: ModuleExportName, declarationName: string, moduleSymbol: Symbol, moduleName: string): void {
+        // for import { "ident" as T } we can provide meaningful error message. (ident is defined locally)
+        // for import { "not ident" as T } we will still look for the locals of the target module symbol, but
+        // since it's impossible to define a local ident with invalid name, we will use the normal message "Module_0_has_no_exported_member_1"

```

Comment:
This comment is a little bit distracting from the point of the function itself. The fact that the impossible cases just result in a missed lookup is not worth mentioning in my opinion, so I'd delete it.